src/metabase/driver/postgres.clj

```diff
@@ -283,24 +283,16 @@
 
 (defn- json-query [identifier nfc-field]
   (letfn [(handle-name [x] (if (number? x) (str x) (name x)))]
-    (let [field-type           (:effective_type nfc-field)
+    (let [field-type           (:database_type nfc-field)
```

Comment:
Dumb question but if we have a database type `text` column that we want to parse as JSON (to database type `json` or whatever it's called) in the future would this not work? I don't think we have a coercion strategy for text -> JSON yet but if we did it would be nice to support that

Reply:
if the text column is assigned semantically to JSON it works fine. problem is that the classifier for assigning semantically to JSON really _is_ a classifier, like a probabilistic thing